### DISCUSSION TITLE
Limits for Incremental Static Regenerator.

### DISCUSSION THREAD

**Discussion by M1ck0** [vercel/next.js#18120] [Help]
I have a question about Incremental Static Regenerator. 

Is there a limit on how many pages can statically be generated? 
Are there any drawbacks to generating, let's say, 200k static pages?
If there are limits, is there a way to optimize this?

Thank you for your time.

**Comment by ljosberinn** [comment]
One limit I can definitely see with that amount is build time. Chances are most hosts won't like that.

**Reply by M1ck0** [reply]
That makes sense. Thank you

**Comment by valse** [comment]
I saw a limit with a single server Node process too: if you have a lot of "same" pages (I mean the same template page, [slug].js) I can't use a `revalidate` bigger then some seconds: in this case the page will never update

**Reply by M1ck0** [reply]
This is good to know. All my pages are in the same template. Thank you for info :)

**Reply by valse** [reply]
I notice it in a single server environment... if you'll use a serverless hosting like vercel you don't have this issue but you'll pay for each revalidate call as single serverless function execution
